frontend/src/components/TransactionsList.tsx

```diff
@@ -13,15 +25,42 @@ function TransactionsList({
   appId,
   showReceiveButton = true,
 }: TransactionsListProps) {
-  const { data: transactions, isLoading } = useTransactions(appId);
+  const [page, setPage] = useState(1);
+  const { data: transactionData, isLoading } = useTransactions(
+    appId,
+    false,
+    LIST_TRANSACTIONS_LIMIT,
+    page
+  );
+  const transactions = transactionData?.transactions || [];
+  const totalCount = transactionData?.totalCount || 0;
+  const totalPages = Math.ceil(totalCount / LIST_TRANSACTIONS_LIMIT);
+
+  const pageNumbers = useMemo(() => {
+    return generatePageNumbers(page, totalPages);
+  }, [page, totalPages]);
 
-  if (isLoading) {
+  useEffect(() => {
+    const el = document.querySelector(".transaction-list");
+    if (el) {
+      el.scrollIntoView({ behavior: "smooth", block: "start" });
+    }
+  }, [page]);
+
+  if (isLoading || !transactionData) {
     return <Loading />;
   }
 
+  const handlePageChange = (newPage: number) => {
+    if (newPage < 1 || newPage > totalPages) {
```

Comment:
how does this happen?

Reply:
I think this was from before when we didn't have pointer-events-none conditionally, will remove, thanks!